File: IOS_swif_study/15_TabBarController_No_storyboard_obj-c/TabBarController_No_Storyboard_objective-c/MainTabBarController.m
//
//  MainTabBarController.m
//  TabBarController_No_Storyboard_objective-c
//
//  Created by Jeff Jeong on 2020/06/02.
//  Copyright © 2020 Tuentuenna. All rights reserved.
//

#import "MainTabBarController.h"
#import "MyViewController.h"

@interface MainTabBarController ()

@end

@implementation MainTabBarController


- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    NSLog(@"MainTabBarController - viewDidLoad() called");
    
    self.view.backgroundColor = [UIColor whiteColor];
    
    UINavigationController * firstNC = [[UINavigationController alloc] initWithRootViewController: [[MyViewController alloc] initWithParams:@"첫번째" bgColor: [UIColor redColor]]];
    
    UINavigationController * secondNC = [[UINavigationController alloc] initWithRootViewController: [[MyViewController alloc] initWithParams:@"두번째" bgColor:[UIColor yellowColor]]];
    
    UINavigationController * thirdNC = [[UINavigationController alloc] initWithRootViewController: [[MyViewController alloc] initWithParams:@"세번째" bgColor:[UIColor orangeColor]]];
    
    UINavigationController * forthNC = [[UINavigationController alloc] initWithRootViewController: [[MyViewController alloc] initWithParams:@"네번째" bgColor:[UIColor greenColor]]];
    
    UINavigationController * fifthNC = [[UINavigationController alloc] initWithRootViewController: [[MyViewController alloc] initWithParams:@"다섯번째" bgColor:[UIColor purpleColor]]];
    
    // 배열준비
    NSMutableArray * vcArray = [NSMutableArray array];
    
    [vcArray addObject:firstNC];
    [vcArray addObject:secondNC];
    [vcArray addObject:thirdNC];
    [vcArray addObject:forthNC];
    [vcArray addObject:fifthNC];
    
    UITabBarItem * firstTabBarItem = [[UITabBarItem alloc] initWithTitle:@"첫번째" image:[UIImage systemImageNamed: @"square.and.arrow.down.fill"]tag:0];
    UITabBarItem * secondTabBarItem = [[UITabBarItem alloc] initWithTitle:@"두번째" image:[UIImage systemImageNamed: @"paperplane.fill"]tag:1];
    UITabBarItem * thirdTabBarItem = [[UITabBarItem alloc] initWithTitle:@"세번째" image:[UIImage systemImageNamed: @"pencil.circle.fill"]tag:2];
    UITabBarItem * forthTabBarItem = [[UITabBarItem alloc] initWithTitle:@"네번째" image:[UIImage systemImageNamed: @"book.fill"]tag:3];
    UITabBarItem * fifthTabBarItem = [[UITabBarItem alloc] initWithTitle:@"다섯번째" image:[UIImage systemImageNamed: @"bookmark.fill"]tag:4];
    
    
    firstNC.tabBarItem = firstTabBarItem;
    secondNC.tabBarItem = secondTabBarItem;
    thirdNC.tabBarItem = thirdTabBarItem;
    forthNC.tabBarItem = forthTabBarItem;
    fifthNC.tabBarItem = fifthTabBarItem;
    
    
    self.viewControllers = vcArray;
    

}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: IOS_swif_study/TabBarController_No_storyboard_Swift/TabBarController_No_storyboard_Swift/MainTabBarController.swift
//
//  MainTabBarController.swift
//  TabBarController_No_storyboard_Swift
//
//  Created by 성정현 on 2022/05/31.
//

import UIKit

class MainTabBarController: UITabBarController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        print("MainTabBarController - viewDidload() called")
        
        let firstNC = UINavigationController.init(rootViewController: MyViewController(title: "첫번째", bgColor: #colorLiteral(red: 0.4745098054, green: 0.8392156959, blue: 0.9764705896, alpha: 1)))
        let secondNC = UINavigationController.init(rootViewController: MyViewController(title: "두번째", bgColor: #colorLiteral(red: 0.5568627715, green: 0.3529411852, blue: 0.9686274529, alpha: 1)))
        let thirdNC = UINavigationController.init(rootViewController: MyViewController(title: "세번쨰", bgColor: #colorLiteral(red: 0.9098039269, green: 0.4784313738, blue: 0.6431372762, alpha: 1)))
        let forthNC = UINavigationController.init(rootViewController: MyViewController(title: "네번쨰", bgColor: #colorLiteral(red: 0.9568627477, green: 0.6588235497, blue: 0.5450980663, alpha: 1)))
        let fifthNC = UINavigationController.init(rootViewController: MyViewController(title: "다섯번째", bgColor: #colorLiteral(red: 0.9764705896, green: 0.850980401, blue: 0.5490196347, alpha: 1)))
        
        self.viewControllers = [firstNC, secondNC, thirdNC, forthNC, fifthNC]
        
        let firstTabBarItem = UITabBarItem(title: "첫번째" , image: UIImage(systemName: "airplayaudio"), tag:0)
        let secondTabBarItem = UITabBarItem(title: "두번째" , image: UIImage(systemName: "airplayvideo"), tag:1)
        let thirdTabBarItem = UITabBarItem(title: "세번째" , image: UIImage(systemName: "arrow.clockwise.icloud.fill"), tag:2)
        let forthTabBarItem = UITabBarItem(title: "네번째" , image: UIImage(systemName: "arrow.down.left.video.fill"), tag:3)
        let fifthTabBarItem = UITabBarItem(title: "다섯번째" , image: UIImage(systemName: "video.fill"), tag:4)
        
        firstNC.tabBarItem = firstTabBarItem
        secondNC.tabBarItem = secondTabBarItem
        thirdNC.tabBarItem = thirdTabBarItem
        forthNC.tabBarItem = forthTabBarItem
        fifthNC.tabBarItem = fifthTabBarItem
        
    }
                                                
}
